tests/integration/ha_tests/test_smoke.py

```diff
@@ -0,0 +1,188 @@
+    # Deploy the charm.
+    async with ops_test.fast_forward():
+        await ops_test.model.deploy(
+            charm,
+            application_name=APPLICATION_NAME,
+            num_units=1,
+            series=CHARM_SERIES,
+            config={"profile": "testing"},
+        )
+
+        # Reducing the update status frequency to speed up the triggering of deferred events.
+        await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+        assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+        primary_name = await get_primary(
+            ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name
+        )
+        assert await is_postgresql_ready(ops_test, primary_name)
+
+        storage_id_str = storage_id(ops_test, primary_name)
+
+        # Check if storage exists after application deployed
+        assert await is_storage_exists(ops_test, storage_id_str)
+
+        await ops_test.model.remove_application(
+            APPLICATION_NAME, block_until_done=True, destroy_storage=True
+        )
+
+        # Check if storage removed after application removal
+        assert not await is_storage_exists(ops_test, storage_id_str)
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_app_force_removal(ops_test: OpsTest, charm: str):
+    """Remove unit with force while storage is alive."""
+    async with ops_test.fast_forward():
+        # Deploy the charm.
+        await ops_test.model.deploy(
+            charm,
+            application_name=APPLICATION_NAME,
+            num_units=1,
+            series=CHARM_SERIES,
+            storage={"pgdata": {"pool": "lxd-btrfs", "size": 8046}},
+            config={"profile": "testing"},
+        )
+
+        # Reducing the update status frequency to speed up the triggering of deferred events.
+        await ops_test.model.set_config({"update-status-hook-interval": "10s"})
+
+        await ops_test.model.wait_for_idle(apps=[APPLICATION_NAME], status="active", timeout=1500)
+        assert ops_test.model.applications[APPLICATION_NAME].units[0].workload_status == "active"
+
+        primary_name = await get_primary(
+            ops_test, ops_test.model.applications[APPLICATION_NAME].units[0].name
+        )
+        assert await is_postgresql_ready(ops_test, primary_name)
+
+        storage_id_str = storage_id(ops_test, primary_name)
+
+        # Check if storage exists after application deployed
+        assert await is_storage_exists(ops_test, storage_id_str)
+
+        # Create test database to check there is no resources conflicts
+        await create_db(ops_test, APPLICATION_NAME, TEST_DATABASE_RELATION_NAME)
+
+        # Destroy charm
+        await ops_test.model.destroy_unit(
+            primary_name, force=True, destroy_storage=False, max_wait=1500
```

Comment:
`destroy_storage` param is only available in python lib juju 3. We should either call the CLI to achieve the same result or make the test run only on juju 3.